packages/wallet/src/services/HandlesTracker.ts

```diff
@@ -0,0 +1,89 @@
+import { Assets, HandleInfo } from '../types';
+import { Cardano } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import {
+  NEVER,
+  Observable,
+  combineLatest,
+  distinctUntilChanged,
+  map,
+  mergeMap,
+  of,
+  shareReplay,
+  withLatestFrom
+} from 'rxjs';
+import { deepEquals, isNotNil } from '@cardano-sdk/util';
+import { sameArrayItems, strictEquals } from './util';
+import uniqBy from 'lodash/uniqBy';
+
+export interface HandlesTrackerProps {
+  utxo$: Observable<Cardano.Utxo[]>;
+  assetInfo$: Observable<Assets>;
+  tip$: Observable<Cardano.Tip>;
+  handlePolicyIds: Cardano.PolicyId[];
+  logger: Logger;
+}
+
+const handleInfoEquals = (a: HandleInfo, b: HandleInfo) =>
```

Comment:
This could be part of 'util' (package or wallet/services/util/equals), and be unit tested separately.
But if you think it only makes sense here, to keep things encapsulated, then it's ok.

Reply:
I'd prefer to keep it there and not export it, not very likely to be used elsewhere. It's tested indirectly with https://github.com/input-output-hk/cardano-js-sdk/pull/752/files#diff-85aa868305ebb433a1a6444ac1039908afa8ca085dc0889066e9ebc44d2afb72R70